client/payments/data/types.ts

```diff
@@ -0,0 +1,32 @@
+export type PaymentGateway = {
+	id: string;
+	title: string;
+	description: string;
+	order: number | string;
+	enabled: boolean;
+	method_title: string;
+	method_description: string;
+	method_supports: Array< string >;
+	settings: Record< string, Record< string, string > >;
+};
+
+export type PluginsState = {
+	paymentGateways: Array< PaymentGateway >;
+	requesting: Record< string, boolean >;
+	errors: Record< string, RestApiError >;
+};
+
+interface RestApiErrorData {
+	status: number;
+}
+
+export type RestApiError = {
+	code: string;
+	data: Partial< RestApiErrorData >;
+	message: string;
+};
+
+export type SelectorKeysWithActions =
+	| 'getPaymentGateways'
+	| 'getPaymentGateway'
+	| 'updatePaymentGateway';
```

Comment:
It might also be wort exporting a `PaymentSelectors` type as we are doing [here with PluginSelectors](https://github.com/woocommerce/woocommerce-admin/blob/main/packages/data/src/plugins/selectors.ts#L67-L72), this way if we use it in our app somewhere we can do:
```
useSelect( (select) => {
  const { ... } = select('wc/payments') as PaymentSelectors;
  // We should now have all the correct selector types available here.
});
```